usr.sbin/inetd/parse_v2.h

```diff
@@ -0,0 +1,10 @@
+#include "inetd.h"
+
+typedef enum parse_v2_result {V2_SUCCESS, V2_SKIP, V2_ERROR} parse_v2_result;
```

Comment:
Looking at all the parsing code, I wonder if it would not have been simpler to use lex and yacc...

Reply:
Yeah Alex looked into this. Three of us took a compiler class last quarter and realized it would probably have been simpler. Even a hand-done scanner/parser might be simpler. Not sure what the best solution would be at this point. The one advantage of the code we have is it just marches across the text without requiring a special stack or anything else. Each key is looked up and just becomes a function pointer, and each key handler reads values one by one at its own discretion instead of being driven by the parser. But I don't know how much that's worth the complexity or verbosity of the code we have.

Reply:
Yes, learning lex and yacc is useful for things like this and are good tools to know. I am not suggesting to re-do the whole thing, but something to keep in mind...